src/main/java/com/lambdaworks/redis/SimpleRedisConnection.java

```diff
@@ -0,0 +1,144 @@
+package com.lambdaworks.redis;
+
+import com.lambdaworks.redis.SimpleSerializers.JavaSerializer;
+import com.lambdaworks.redis.codec.ByteArrayCodec;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * @author <a href="mailto:a.abdelfatah@live.com">Ahmed Kamal</a>
+ * @since 12.01.16 09:17
+ */
+
+/**
+ * A simple redis connection for caching and retrieval from redis
+ */
+public class SimpleRedisConnection {
+
+    private final RedisClient redisClient;
+    protected static final InternalLogger logger = InternalLoggerFactory.getInstance(SimpleRedisConnection.class);
+
+    public SimpleRedisConnection(RedisClient redisClient) {
+        this.redisClient = redisClient;
+    }
+
+    /**
+     * Cache generic value in redis (The class of the value should implement Serializable)
+     * <br/>
+     * Sets field in the hash stored at key to value. If key does not exist, a new key holding a hash is created.
+     * <br/>
+     * If field already exists in the hash, it is overwritten.
+     *
+     * @param key
+     * @param field
+     * @param value
+     * @param <T>
+     * @return true if field is a new field in the hash and value was set.
+     * <br/> false if field already exists in the hash and the value was updated.
+     */
+    public <T> boolean cache(String key, String field, T value) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            byte[] serializedValues = new JavaSerializer().serializeObject(value);
+            redisCommands.hset(key.getBytes(), field.getBytes(), serializedValues);
+
+            return true;
+        } catch (Exception e) {
+            logger.error("Exception Occurred while caching a value in redis", e);
```

Comment:
Catching and logging an Exception hides actually that an exception happened. The better pattern is to not catch any Exceptions at all and to propagate the Exception to the caller. In general, if you can handle the exception (just logging is no Exception handling), then catch the Exception otherwise give the Exception to the caller. Wrapping Exceptions into own Exception types is also a common pattern for Exceptions but then you need to follow this pattern throughout the library.